pkg/agent/util/net_windows_test.go

```diff
@@ -0,0 +1,1342 @@
+					Mask: net.CIDRMask(32, 32),
+				},
+			},
+			commandOut:           "",
+			commandErr:           fmt.Errorf("interface already exists"),
+			testNetInterfaceName: "0",
+			wantErr:              fmt.Errorf("failed to remove address 8.8.8.8/32 from interface 0: interface already exists"),
+		},
+		{
+			name: "Link Addr IPv6 Not Supported",
+			idx:  0,
+			ipNets: []*net.IPNet{
+				{
+					IP:   net.IPv6zero,
+					Mask: net.CIDRMask(128, 128),
+				},
+			},
+			commandOut:           "",
+			commandErr:           nil,
+			testNetInterfaceName: "0",
+			wantErr:              nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runCommand = func(cmd string) (string, error) {
+				return tc.commandOut, tc.commandErr
+			}
+			netInterfaceByIndex = func(index int) (*net.Interface, error) {
+				testNetInterface := generateNetInterface(strconv.Itoa(index))
+				tc.testNetInterface = testNetInterface
+				return (*net.Interface)(testNetInterface), tc.testNetInterfaceErr
+			}
+			netInterfaceAddrs = func(i *net.Interface) ([]net.Addr, error) {
+				if i != nil {
+					netAddrs, _ := tc.testNetInterface.Addrs()
+					return netAddrs, tc.testInterfaceAddrsErr
+				}
+				return nil, tc.testInterfaceAddrsErr
+			}
+			defer func() {
+				runCommand = ps.RunCommand
+				netInterfaceByIndex = net.InterfaceByIndex
+				netInterfaceAddrs = (*net.Interface).Addrs
+			}()
+			gotErr := ConfigureLinkAddresses(tc.idx, tc.ipNets)
+			assert.Equal(t, tc.wantErr, gotErr)
+		})
+	}
+}
+
+func TestSetAdapterMACAddress(t *testing.T) {
+	testMacAddr, _ := net.ParseMAC("aa:bb:cc:dd:ee:ff")
+	runCommand = func(cmd string) (string, error) {
+		return "success", nil
+	}
+	defer func() {
+		runCommand = ps.RunCommand
+	}()
+	gotErr := SetAdapterMACAddress("test-adapter", &testMacAddr)
+	assert.NoError(t, gotErr)
+	testErr := fmt.Errorf("invalid")
+	runCommand = func(cmd string) (string, error) {
+		return "", testErr
+	}
+	gotErr = SetAdapterMACAddress("test-adapter", &testMacAddr)
+	assert.Equal(t, testErr, gotErr)
+}
+
+func TestWindowsHyperVEnabled(t *testing.T) {
+	runCommand = func(cmd string) (string, error) {
+		return "Enabled_HyperV", nil
+	}
+	defer func() {
+		runCommand = ps.RunCommand
+	}()
+	gotResult, gotErr := WindowsHyperVEnabled()
+	assert.True(t, gotResult)
+	assert.NoError(t, gotErr)
+	testErr := fmt.Errorf("invalid")
+	runCommand = func(cmd string) (string, error) {
+		return "fail", testErr
+	}
+	gotResult, gotErr = WindowsHyperVEnabled()
+	assert.False(t, gotResult)
+	assert.Equal(t, testErr, gotErr)
+}
+
+func TestPrepareHNSNetwork(t *testing.T) {
+	testMacAddr, _ := net.ParseMAC("aa:bb:cc:dd:ee:ff")
+	gw, subnet, _ := net.ParseCIDR("8.8.8.8/32")
+	routes := []Route{{
+		LinkIndex:         0,
+		DestinationSubnet: subnet,
+		GatewayAddress:    gw,
+		RouteMetric:       MetricDefault,
+	}}
+	testRoutes := make([]interface{}, len(routes))
+	for i, v := range routes {
+		testRoutes[i] = v
+	}
```

Comment:
I see there are similar codes to build up `testRoutes` in both linux and windows test files, you may define a new function like `createTestRoutes` to reuse it.

Reply:
Defined function in both linux and windows test files, because the `Route`s are of different types, especially `Route` in windows is defined in `net_windows.go`.